### PR TITLE
DM-44647: Move pipeline-dot build from cmdLineFwk to builder

### PR DESCRIPTION
**Pull request by leeskelvin**
This commit moves the pipeline-dot build logic from cmdLineFwk into cli/script/build.py.
As part of this commit, a switch to using the same back-end display args parser as --show pipeline-graph is also made.

## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/mpexec/cmdLineFwk.py
Comment (timj) [inline python/lsst/ctrl/mpexec/cmdLineFwk.py:586]: I am now thinking how this PR relates to #159 given that this is the only usage of pipeline2dot and that PR was removing it completely.

Comment (TallJimbo) [reply python/lsst/ctrl/mpexec/cmdLineFwk.py:586]: I think this fully supersedes #159.  I'm also planning to RFC dropping support for QuantumGraph GraphViz outputs, since my understanding is that those are always too big for `dot` to render in practice anyway, and then we can get rid of `dotTools.py` entirely.

**Comment by leeskelvin** [comment]
I've removed the `DO NOT MERGE` commit from this PR. Note that the `mypy` check now fails. After this ticket has been approved, I'll merge the `pipe_base` branch to `main` first, then re-run the checks on this PR to ensure that all checks are successful prior to merge.

**Review by arunkannawadi — APPROVED** [review]

File: requirements.txt
Comment (timj) [inline requirements.txt:12]: Hopefully this can be removed now.

Comment (leeskelvin) [reply requirements.txt]: Thanks! Commit dropped.

**Pull request merged**
